projects/ngrx.io/content/guide/signals/signal-store/testing.md

```diff
@@ -0,0 +1,676 @@
+
+    tick(50);
+    expect(store.movies()).toEqual([{ id: 2, name: 'Jurassic Park' }]);
+    expect(store.loading()).toBe(false);
+  }));
+});
+
+</code-example>
+
+It is important to account for the glitch-free effect when using Signals. The `rxMethod` relies on `effect`, which may need to be triggered manually through `TestBed.flushEffects()`.
+
+If the mocked `MovieService` operates synchronously, the following test will fail unless `TestBed.flushEffects()` is called.
+
+<code-example header="movies.store.spec.ts">
+
+describe('MoviesStore', () => {
+  // ... beforeEach, and afterEach omitted
+
+  it('should depend on flushEffects because of synchronous execution', () => {
+    const moviesService = {
+      load: jest.fn((studio: string) => of([studio === 'Warner Bros' ? { id: 1, name: 'Harry Potter' } : { id: 2, name: 'Jurassic Park' }])),
+    };
+
+    TestBed.configureTestingModule({
+      providers: [
+        {
+          provide: MoviesService,
+          useValue: moviesService,
+        },
+      ],
+    });
+
+    const store = TestBed.inject(MoviesStore);
+    const studio = signal('Warner Bros');
+    store.load(studio);
+    TestBed.flushEffects(); // required
+    expect(store.movies()).toEqual([{ id: 1, name: 'Harry Potter' }]);
+
+    studio.set('Universal');
+    TestBed.flushEffects(); // required
+    expect(store.movies()).toEqual([{ id: 2, name: 'Jurassic Park' }]);
+  });
+});
+
+</code-example>
+
+## Mocking the SignalStore
+
+What applies to testing the SignalStore also applies to mocking it. The SignalStore functions like any other service, meaning it can be mocked using the same tools and techniques applied to other services.
+
+The `MovieComponent` utilizes the `MoviesStore` to display movies:
+
+<code-example header="movies.component.ts">
+
+@Component({
+  selector: 'app-movies',
+  template: ` <input type="text" [(ngModel)]="studio" [disabled]="store.loading()" placeholder="Name of Studio" />
+    <ul>
+      @for (movie of store.movies(); track movie.id) {
+      <p>{{ movie.id }}: {{ movie.name }}</p>
+      }
+    </ul>`,
+  standalone: true,
+  imports: [FormsModule],
+})
+export class MoviesComponent {
+  protected studio = signal('');
+  protected readonly store = inject(MoviesStore);
+
+  constructor() {
+    this.store.load(this.studio);
+  }
+}
+
+</code-example>
+
+### Native Mocking
+
+<code-example header="movies.component.spec.ts">
+
+it('should show movies (native Jest)', () => {
+  const load = jest.fn<void, [Signal<string>]>();
+
+  const moviesStore = {
+    movies: signal(new Array<Movie>()),
+    loading: signal(false),
+    load,
+  };
+
+  TestBed.configureTestingModule({
+    imports: [MoviesComponent],
+    providers: [
+      {
+        provide: MoviesStore,
+        useValue: moviesStore,
+      },
+    ],
+  });
+
+  const fixture = TestBed.createComponent(MoviesComponent);
+  fixture.autoDetectChanges(true);
+
+  const studio = load.mock.calls[0][0];
```

Comment:
I think might be wrong.  Only one `[0]` ?

Reply:
`[0][0]` means the first parameter of the first call.

Reply:
Yes, you are right. `mock.calls` returns an array of arrays.